scripts/data/gen-nodetime/src/relayer/lib/xrelayer.ts

```diff
@@ -0,0 +1,490 @@
+				);
+			}
+			if (
+				this.config.chains &&
+				this.config.chains.find(
+					(x) => x.chainId == chain.chainId && x.rpcAddr == chain.rpcAddr
+				)
+			) {
+				return { id: chain.chainId };
+			} else {
+				this.config.chains.push(chain);
+				return { id: chain.chainId };
+			}
+		} catch (e) {
+			throw new Error("Could not setup chain: " + e);
+		}
+	}
+	public createPath([srcID, dstID, options]: [
+		string,
+		string,
+		ConnectOptions
+	]): Path {
+		try {
+			let path = {
+				id: srcID + "-" + dstID,
+				isLinked: false,
+				src: {
+					chainID: srcID,
+					portID: options.sourcePort,
+				},
+				dst: {
+					chainID: dstID,
+					portID: options.targetPort,
+				},
+			};
+			if (!this.config.paths) {
+				this.config.paths = [];
+			}
+			this.config.paths.push({ path, options });
+			return path;
+		} catch (e) {
+			throw new Error("Could not create path:" + e);
+		}
+	}
+	public getPath([id]: [string]): Path {
+		if (this.config.paths) {
+			let path = this.config.paths.find((x) => x.path.id == id);
+			if (path) {
+				return path.path;
+			} else {
+				throw new Error("Path does not exist");
+			}
+		} else {
+			throw new Error("Path does not exist");
+		}
+	}
+	public listPaths(): Path[] {
+		if (this.config.paths) {
+			let paths = this.config.paths.map((x) => x.path);
+			return paths;
+		} else {
+			throw new Error("No paths defined");
+		}
+	}
+	public async getDefaultAccount([chainID]: string[]): Promise<Account> {
+		const chain = this.chainById(chainID);
+		if (chain) {
+			let client = await this.getIBCClient(chain);
+			return {
+				address: client.senderAddress,
+			};
+		} else {
+			throw new Error("Chain not found: " + chainID);
+		}
+	}
+	public async getDefaultAccountBalance([chainID]: string[]): Promise<Coin[]> {
+		const chain = this.chainById(chainID);
+		if (chain) {
+			let client = await this.getIBCClient(chain);
+			return await client.query.bank.allBalances(client.senderAddress);
+		} else {
+			throw new Error("Chain not found: " + chainID);
+		}
+	}
+	public async link(paths: string[]): Promise<LinkResponse> {
+		if (this.config.paths) {
+			let response: LinkResponse = {
+				linkedPaths: [],
+				alreadyLinkedPaths: [],
+			};
+			for (let pathName of paths) {
+				const path = this.pathById(pathName);
+				if (path?.path.isLinked) {
+					response.alreadyLinkedPaths.push(pathName);
+				} else {
+					try {
+						await this.createLink(path);
+						response.linkedPaths.push(pathName);
+					} catch (e) {
+						throw new Error("Could not link path: " + pathName + ": " + e);
+					}
+				}
+			}
+			return response;
+		} else {
+			throw new Error("No paths defined");
+		}
+	}
+	public async start(paths: string[]): Promise<StartResponse> {
+		if (this.config.paths) {
+			for (let pathName of paths) {
+				const path = this.pathById(pathName);
+				if (path?.path.isLinked) {
+					const link = await this.getLink(path);
+					this.relayers.set(
```

Comment:
Why setInterval's id added to relayers map? it's never accessed after in the source code.

Reply:
Just holding a reference for potential improvements/refactoring in the future, possibly allowing a command to stop/pause/restart a single path etc.